src/overlays/actors/ovl_Boss_Ganon2/z_boss_ganon2.c

```diff
@@ -278,116 +304,3125 @@ static ColliderJntSphInit D_80907014 = {
+            } else {
+                phi_f20 = 1.0f - ((i - 7) * 0.2333333f);
+            }
+
+            func_800D1FD4(&globalCtx->mf_11DA0);
+            Matrix_Scale(200.0f * phi_f20, 200.0f * phi_f20, 1.0f, MTXMODE_APPLY);
+            Matrix_RotateZ(Rand_ZeroFloat(2.0f * M_PI), MTXMODE_APPLY);
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 5721),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_XLU_DISP++, SEGMENTED_TO_VIRTUAL(ovl_Boss_Ganon2_DL_00B378));
+        }
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 5725);
+}
+
+void BossGanon2_PostLimbDraw2(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, void* thisx) {
+    s8 temp_v1 = D_80907144[limbIndex];
+    BossGanon2* this = THIS;
+
+    if (temp_v1 >= 0) {
+        Matrix_MultVec3f(&D_80906D60, &this->unk_234[temp_v1]);
+    }
+    if (limbIndex == 11) {
+        OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 5749);
+
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 5752),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_XLU_DISP++, SEGMENTED_TO_VIRTUAL(object_ganon_DL_00BE90));
+
+        CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 5754);
+    } else if (limbIndex == 10) {
+        Matrix_MultVec3f(&D_80907164, &this->unk_1B8);
+    }
+}
+
+void func_80905674(BossGanon2* this, GlobalContext* globalCtx) {
+    s32 pad;
+
+    if (this->unk_380 > 0.0f) {
+        OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 5772);
+
+        Matrix_Push();
+        gDPPipeSync(POLY_XLU_DISP++);
+        gSPSegment(POLY_XLU_DISP++, 0x08,
+                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, this->unk_19C * -8, 0, 32, 64, 1, this->unk_19C * -4,
+                                    this->unk_19C * -8, 32, 32));
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 170, (s16)this->unk_37C);
+        gDPSetEnvColor(POLY_XLU_DISP++, 255, 200, 0, 128);
+        Matrix_Translate(sZelda->actor.world.pos.x + 100.0f, sZelda->actor.world.pos.y + 35.0f + 7.0f,
+                         sZelda->actor.world.pos.z - 100.0f, MTXMODE_NEW);
+        Matrix_RotateY(-M_PI / 4.0f, MTXMODE_APPLY);
+        Matrix_Scale(0.040000003f, 0.040000003f, this->unk_380, MTXMODE_APPLY);
+        Matrix_RotateX(M_PI / 2.0f, MTXMODE_APPLY);
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 5814),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_XLU_DISP++, SEGMENTED_TO_VIRTUAL(ovl_Boss_Ganon2_DL_00EC40));
+        Matrix_Pop();
+
+        CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 5817);
+    }
+}
+
+void BossGanon2_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    u8* shadowTexture = Graph_Alloc(globalCtx->state.gfxCtx, 4096);
```

Comment:
```suggestion
    void* shadowTexture = Graph_Alloc(globalCtx->state.gfxCtx, 4096);
```
We usually use void* for textures if this works.